Contributing.md

```diff
@@ -1,30 +1,471 @@
+
+The Tari release process draws inspiration from the Rust [compiler release process] and borrows some of its key 
+ideas.
+
+The process is well-suited for a large open-source project that desires a regular release schedule but also needs 
+the flexibility of having features being "in development" over several cycles.
+
+## Release branches
+
+Post mainnet, there are 4 release branches in the Tari repo:
+
+* `mainnet` - the code on this branch represents the latest "official release". Almost every commit on this branch 
+  will be tagged with a version number.
+* `development` - this is the cuttingest-of-edge branch. Almost all new code comes in via the `development` branch.
+* `nextnet` - the code on this branch will become the `mainnet` branch on the next release. Feature code from 
+  `dibbler` will be present here, but if they are not yet ready for release, they will be disabled behind a feature 
+  gate.
+* `dibbler` - this branch contains experimental features that might not work, be rigorously tested, or even break 
+  consensus if the code were to run on mainnet.
+
+Almost all new code enters the codebase via the `development` branch. Hotfixes and urgent updates can be made on 
+other branches from time to time. These changes will then be cherry-picked into the other branches to keep 
+the code base consistent.
+
+## Testnets
+
+There will also be three testnets in addition to mainnet:
+
+* `mainnet` - the "real-money" network.
+* `stagenet` - a testnet version of mainnet. Nodes on stagenet should all be running code built from the `mainnet`
+  branch to replicate mainnet as close as possible. The base node version on `stagenet` will always match
+  the latest `mainnet` release. Network resets (i.e. a new genesis block) should be very rare.
+* `nextnet` - nodes on nextnet should be running code compiled from the `nextnet` branch. This is the last chance to
+  battle-test new features before they hit mainnet. Network resets should be somewhat rare.
+* `dibbler` - The dibbler network is the most experimental of the testnets. Things can break here. We hope they 
+  won't, obviously, but the dibbler network is where we should be catching any and all major issues _before_ 
+  the feature heads to nextnet. Nodes should be running code built from the `dibbler` branch. Network resets could 
+  occur fairly frequently, perhaps several times a year.
+
+## Feature flags
+
+When developing a new feature in Tari, it will initially be placed behind a feature flag -- similar to how Rust 
+nightly features are gated. This offers the comfort and flexibility that should a feature not be ready in the first 
+6-8 months of development (i.e. by the time it first hits the `mainnet` branch), it will not be active on the 
+mainnet, since the feature will not be active.
+
+The name of the feature flag is assigned in the tracking issue 
+of the RFC that describes the feature. Once the feature is complete and ready for testing, it can be enabled in the 
```

Comment:
Given all the branches, it is probably going to be worth splitting RFC's into their own repo, otherwise the related RFC might be in a different branch to the feature being implemented

Reply:
Possibly.

Reply:
Will each feature be preceded by some portion of an RFC or collection of RFCs? If so we could always have the RFC(s) debated and merged into development before work on the feature starts.

In any case, I think we would need a working document in development that lists and describes all features being worked on and precedes actual commitments on any specific feature.

Reply:
For the most part, there will be a mapping of feature to RFC. 
The idea being that if an idea is big enough to warrant an RFC, it's big enough to earn a feature gate.